enhancements/two-nodes-openshift/2no.md

```diff
@@ -0,0 +1,507 @@
+---
+title: 2no
+authors:
+  - "@mshitrit"
+reviewers:
+  - "@rwsu"
+  - "@fabbione"
+  - "@carbonin"
+  - "@thomasjungblut"
+  - "@brandisher"
+  - "@DanielFroehlich"
+  - "@jerpeter1"
+  - "@slintes"
+  - "@beekhof"
+  - "@eranco74"
+  - "@yuqi-zhang"
+  - "@gamado"
+  - "@razo7"
+  - "@frajamomo"
+  - "@clobrano"
+
+approvers:
+  - "@rwsu"
+  - "@fabbione"
+  - "@carbonin"
+  - "@thomasjungblut"
+  - "@brandisher"
+  - "@DanielFroehlich"
+  - "@jerpeter1"
+  - "@slintes"
+  - "@beekhof"
+  - "@eranco74"
+  - "@yuqi-zhang"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "@jerpeter1"
+creation-date: 2024-09-05
+last-updated: 2024-09-22
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-1514
+---
+
+# Two Nodes Openshift (2NO) - Control Plane Availability
+
+## Terms
+
+RHEL-HA - a general purpose clustering stack shipped by Red Hat (and others) primarily consisting of Corosync and Pacemaker.  Known to be in use by airports, financial exchanges, and defense organizations, as well as used on trains, satellites, and expeditions to Mars.
+
+Corosync - a Red Hat led [open-source project](https://corosync.github.io/corosync/) that provides a consistent view of cluster membership, reliable ordered messaging, and flexible quorum capabilities.  
+
+Pacemaker - a Red Hat led [open-source project](https://clusterlabs.org/pacemaker/doc/) that works in conjunction with Corosync to provide general purpose fault tolerance and automatic failover for critical services and applications.
+
+Resource Agent - A resource agent is an executable that manages a cluster resource. No formal definition of a cluster resource exists, other than "anything a cluster manages is a resource." Cluster resources can be as diverse as IP addresses, file systems, database services, and entire virtual machines - to name just a few examples.
+<br>[more context here](https://github.com/ClusterLabs/resource-agents/blob/main/doc/dev-guides/ra-dev-guide.asc)
+
+Fencing - the process of “somehow” isolating or powering off malfunctioning or unresponsive nodes to prevent them from causing further harm or interference with the rest of the cluster.
```

Comment:
Controlling the power state is different from isolating the node, but both are mechanisms to prevent corruption or the creation of divergent datasets.

Reply:
ok

Reply:
resolved